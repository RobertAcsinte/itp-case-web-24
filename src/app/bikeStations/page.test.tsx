import { afterAll, afterEach, beforeAll, expect, it, } from 'vitest';
import { render } from "@testing-library/react";
import Bikes from './page';
import { setupServer } from 'msw/node'
import { HttpResponse, http } from 'msw'


const station = [
    {
        results: [
            {
                id: 1,
                name: 'Station 1',
                bikes_in_use: 2,
                bikes_available: 3
            }
        ]
    },
    {
        results: [
            {
                id: 2,
                name: 'Station 2',
                bikes_in_use: 7,
                bikes_available: 7
            }
        ]
    }
]
  
export const restHandlers = [
    http.get('https://data.stad.gent/api/explore/v2.1/catalog/datasets/blue-bike-deelfietsen-gent-dampoort/records?select=id%2C%20name%2C%20bikes_in_use%2C%20bikes_available', () => {
        return HttpResponse.json(station[0])
    }),
    http.get('https://data.stad.gent/api/explore/v2.1/catalog/datasets/blue-bike-deelfietsen-gent-sint-pieters-m-hendrikaplein/records?select=id%2C%20name%2C%20bikes_in_use%2C%20bikes_available', () => {
        return HttpResponse.json(station[1])
    }),
]

const server = setupServer(...restHandlers)

beforeAll(() => server.listen())

afterAll(() => server.close())

afterEach(() => server.resetHandlers())

it('Snapshot Bike Station Page', async () => {
    const jsx = await Bikes();
    const result = render(jsx)
    expect(result).toMatchSnapshot()
})