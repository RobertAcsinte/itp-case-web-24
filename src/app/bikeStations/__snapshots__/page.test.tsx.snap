// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Bike Station Page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="wrapper"
      >
        <div
          class="title-container"
        >
          <a
            aria-label="Link to go back to the home page"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-left "
              data-icon="arrow-left"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
                fill="currentColor"
              />
            </svg>
          </a>
          <h1>
            Bike Stations
          </h1>
        </div>
        <section>
          <div
            class="_container_f28ee5"
          >
            <div
              class="_titleWrapper_f28ee5"
            >
              <h2>
                Station 1
              </h2>
              <button
                aria-label="Copy station title to clipboard"
                aria-live="polite"
              >
                Copy to clipboard
              </button>
            </div>
            <p>
              <b>
                Bikes available:
              </b>
               
              3
            </p>
            <p>
              <b>
                Total bikes:
              </b>
               
              5
            </p>
          </div>
        </section>
        <section>
          <div
            class="_container_f28ee5"
          >
            <div
              class="_titleWrapper_f28ee5"
            >
              <h2>
                Station 2
              </h2>
              <button
                aria-label="Copy station title to clipboard"
                aria-live="polite"
              >
                Copy to clipboard
              </button>
            </div>
            <p>
              <b>
                Bikes available:
              </b>
               
              7
            </p>
            <p>
              <b>
                Total bikes:
              </b>
               
              14
            </p>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="wrapper"
    >
      <div
        class="title-container"
      >
        <a
          aria-label="Link to go back to the home page"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-left "
            data-icon="arrow-left"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              fill="currentColor"
            />
          </svg>
        </a>
        <h1>
          Bike Stations
        </h1>
      </div>
      <section>
        <div
          class="_container_f28ee5"
        >
          <div
            class="_titleWrapper_f28ee5"
          >
            <h2>
              Station 1
            </h2>
            <button
              aria-label="Copy station title to clipboard"
              aria-live="polite"
            >
              Copy to clipboard
            </button>
          </div>
          <p>
            <b>
              Bikes available:
            </b>
             
            3
          </p>
          <p>
            <b>
              Total bikes:
            </b>
             
            5
          </p>
        </div>
      </section>
      <section>
        <div
          class="_container_f28ee5"
        >
          <div
            class="_titleWrapper_f28ee5"
          >
            <h2>
              Station 2
            </h2>
            <button
              aria-label="Copy station title to clipboard"
              aria-live="polite"
            >
              Copy to clipboard
            </button>
          </div>
          <p>
            <b>
              Bikes available:
            </b>
             
            7
          </p>
          <p>
            <b>
              Total bikes:
            </b>
             
            14
          </p>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
